src/snowflake/snowpark/modin/pandas/utils.py

```diff
@@ -568,17 +570,43 @@ def extract_validate_and_try_convert_named_aggs_from_kwargs(
                 if value[0] not in columns:
                     raise KeyError(f"Column(s) ['{value[0]}'] do not exist")
 
-            if value[0] in named_aggs:
-                if not isinstance(named_aggs[value[0]], list):
-                    named_aggs[value[0]] = [named_aggs[value[0]]]
-                named_aggs[value[0]] += [
-                    AggFuncWithLabel(func=value[1], pandas_label=key)
-                ]
+            # This function converts our named aggregations dictionary from a mapping of
+            # new_label -> tuple[column_name, agg_func] to a mapping of
+            # column_name -> tuple[agg_func, new_label] in order to process
```

Comment:
Not sure we need to do this here? it seems that the difference here is that when the aggfunc is in the format with new_label -> tuple[column_name, agg_func], the new_label is treated as new row label in regular aggregation, and new column label in groupby agg, we should be able to handle this fairly straightforward at query compiler layer, right?

Reply:
The issue is the ordering change. Since the order of the aggregations is not preserved by this dictionary, we need to reorder at the end of the operation. We can do so cheaply for groupby_agg because the new aggregations are the names of the columns - we can just swap the order of the data_column_pandas_labels + data_column_snowflake_quoted_identifiers before making the InternalFrame, and have the correct ordering reflected. For the agg though, the new labels are the index, so reordering is expensive, and so we need a different format of the map that preserves the order of the named aggregations.

Reply:
Can add a code comment to clarify.

Reply:
Resolved via passing order explicitly, and reordering query compilers prior to concatenation.